ts/smelter-node/src/offline/compositor.ts

```diff
@@ -33,9 +38,34 @@ export default class OfflineSmelter {
     });
   }
 
-  public async registerInput(inputId: string, request: RegisterInput) {
-    await this.scheduler.run(async () => {
-      await this.coreSmelter.registerInput(inputId, request);
+  public async registerInput(
+    inputId: string,
+    request: Extract<RegisterInput, { type: 'whip' }>
+  ): Promise<RegisterWhipInputResponse>;
+
+  public async registerInput(
+    inputId: string,
+    request: Extract<RegisterInput, { type: 'mp4' }>
+  ): Promise<RegisterMp4InputResponse>;
+
+  public async registerInput(inputId: string, request: RegisterInput): Promise<object>;
+
+  public async registerInput(inputId: string, request: RegisterInput): Promise<object> {
+    return await this.scheduler.run(async () => {
+      let result = await this.coreSmelter.registerInput(inputId, request);
+      if (request.type === 'mp4') {
+        return {
+          videoDurationMs: result.video_duration_ms,
+          audioDurationMs: result.audio_duration_ms,
+        };
+      } else if (request.type === 'whip') {
+        return {
+          bearerToken: result.bearer_token,
+          endpointRoute: `/whip/${encodeURIComponent(inputId)}`,
```

Comment:
is there a way to handle that in core? If not I would still prefer to extract conversion logic like this to outside of file like this. It mixes to much high level API with simple low level type conversion